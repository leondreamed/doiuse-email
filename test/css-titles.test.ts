import { describe, expect, test } from 'vitest';

import {
  atRuleTitles,
  functionTitles,
  keywordTitles,
  propertyTitles,
  propertyValuePairTitles,
  psuedoSelectorTitles,
  selectorTitles,
  unitTitles
} from '../src/utils/titles/css';

const expectEqualTitles = (titles: Record<string, unknown>) =>
  expect(Object.keys(titles).sort()).toMatchSnapshot();

describe('css titles', () => {
  test('property titles', () => {
    expectEqualTitles(propertyTitles);
  });

  test('at rule titles', () => {
    expectEqualTitles(atRuleTitles);
  });

  test('property value pair titles', () => {
    expectEqualTitles(propertyValuePairTitles);
  });

  test('unit titles', () => {
    expectEqualTitles(unitTitles);
  });

  test('selector titles', () => {
    expectEqualTitles(selectorTitles);
  });

  test('function titles', () => {
    expectEqualTitles(functionTitles);
  });

  test('pseudo selector titles', () => {
    expectEqualTitles(psuedoSelectorTitles);
  });

  test('keyword titles', () => {
    expectEqualTitles(keywordTitles);
  });
});
