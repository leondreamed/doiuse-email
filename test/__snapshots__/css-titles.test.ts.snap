// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`css titles > at rule titles 1`] = `
[
  "@font-face",
  "@import",
  "@keyframes",
  "@media",
  "@media (-webkit-device-pixel-ratio)",
  "@media (hover), @media (any-hover)",
  "@media (orientation)",
  "@media (prefers-color-scheme)",
  "@media (prefers-reduced-motion)",
  "@supports",
]
`;

exports[`css titles > function titles 1`] = `
[
  "CSS Variables (Custom Properties)",
  "CSS calc() function",
  "clamp()",
  "conic-gradient()",
  "lang()",
  "lch(), oklch(), lab(), oklab()",
  "light-dark()",
  "linear-gradient()",
  "max()",
  "min()",
  "radial-gradient()",
  "rgb()",
  "rgba()",
]
`;

exports[`css titles > keyword titles 1`] = `
[
  "!important keyword",
]
`;

exports[`css titles > property titles 1`] = `
[
  "accent-color",
  "align-items",
  "animation",
  "aspect-ratio",
  "backdrop-filter",
  "background",
  "background-blend-mode",
  "background-clip",
  "background-color",
  "background-image",
  "background-origin",
  "background-position",
  "background-repeat",
  "background-size",
  "border",
  "border-collapse",
  "border-image",
  "border-radius",
  "border-spacing",
  "box-shadow",
  "box-sizing",
  "caption-side",
  "clip-path",
  "column-count",
  "direction",
  "display",
  "empty-cells",
  "filter",
  "float",
  "font shorthand",
  "font-kerning",
  "font-size",
  "font-stretch",
  "font-weight",
  "height property",
  "hyphenate-character",
  "hyphenate-limit-chars",
  "hyphens",
  "inline-size ",
  "inset",
  "justify-content",
  "left, right, top, bottom",
  "letter-spacing",
  "line-height",
  "list-style",
  "list-style-image",
  "list-style-position",
  "list-style-type",
  "margin",
  "max-block-size",
  "max-height property",
  "max-inline-size",
  "max-width",
  "min-block-size",
  "min-height property",
  "min-inline-size",
  "min-width property",
  "mix-blend-mode",
  "object-fit",
  "object-position",
  "opacity",
  "orphans",
  "outline",
  "outline-offset",
  "overflow",
  "overflow-wrap",
  "padding",
  "position",
  "resize",
  "scroll-snap",
  "shape-margin",
  "shape-outside",
  "tab-size",
  "table-layout",
  "text-align",
  "text-align-last",
  "text-decoration",
  "text-decoration-color",
  "text-decoration-line",
  "text-decoration-skip-ink",
  "text-decoration-style",
  "text-decoration-thickness",
  "text-emphasis",
  "text-emphasis-position",
  "text-indent",
  "text-justify",
  "text-orientation",
  "text-overflow",
  "text-shadow",
  "text-transform",
  "text-underline-offset",
  "text-underline-position",
  "text-wrap",
  "transform",
  "transition",
  "user-select",
  "vertical-align",
  "visibility",
  "white-space",
  "white-space-collapse",
  "widows",
  "width property",
  "word-break",
  "word-spacing",
  "writing-mode",
  "z-index",
]
`;

exports[`css titles > property value pair titles 1`] = `
[
  "display:flex",
  "display:grid",
  "display:none",
  "flex-direction:column",
  "flex-wrap: wrap",
]
`;

exports[`css titles > pseudo selector titles 1`] = `
[
  "::after",
  "::before",
  "::first-letter",
  "::first-line",
  "::marker",
  "::placeholder",
  ":active",
  ":checked",
  ":first-child",
  ":first-of-type",
  ":focus",
  ":has()",
  ":hover",
  ":last-child",
  ":last-of-type",
  ":link",
  ":not",
  ":nth-child",
  ":nth-last-child",
  ":nth-last-of-type",
  ":nth-of-type",
  ":only-child",
  ":only-of-type",
  ":target",
  ":visited",
]
`;

exports[`css titles > selector titles 1`] = `
[
  "Adjacent sibling combinator",
  "Attribute selector",
  "Chaining selectors",
  "Child combinator",
  "Class selector",
  "Descendant combinator",
  "General sibling combinator",
  "Grouping selectors",
  "ID selector",
  "Type selector",
  "Universal selector *",
]
`;

exports[`css titles > unit titles 1`] = `
[
  "% unit",
  "ch unit",
  "cm unit",
  "em unit",
  "ex unit",
  "in unit",
  "initial unit",
  "mm unit",
  "pc unit",
  "pt unit",
  "px unit",
  "rem unit",
  "vh unit",
  "vmax unit",
  "vmin unit",
  "vw unit",
]
`;
